### ISSUE TITLE
Intellisense won't allow `__builtin_expect` in ctor of constexpr value

### ISSUE THREAD

**Issue by tonyelewis** [microsoft/vscode-cpptools#2824]
**Type: LanguageService**

Thanks very much for all your work on this extension. I really love using both it and VSCode.

Strict Intellisense on this source code:

~~~cpp
struct x {
   constexpr x(int val) {
      [[maybe_unused]] const auto mu = __builtin_expect( val, 0 );
   }
};

[[maybe_unused]] constexpr x x1{ 3 };
~~~

&hellip;puts a red-squiggle under the `x1` on the last line, with mouseover message:

~~~no-highlight
expression must have a constant value -- cannot call non-constexpr function "__builtin_expect" (declared implicitly)
constexpr x x1
~~~

&hellip;but GCC and Clang are happy with it:

~~~sh
g++     -Werror -Wall -Wextra -pedantic -std=c++17 -fsyntax-only a.cpp
clang++ -Werror -Wall -Wextra -pedantic -std=c++17 -fsyntax-only a.cpp
~~~

I've come across this through using the [Microsoft GSL](https://github.com/Microsoft/GSL) library. I can work-around the issue by adding a `GSL_UNENFORCED_ON_CONTRACT_VIOLATION` define in `c_cpp_properties.json`. But it'd be good to get this fixed in VSCode.

Repro steps:
1. Open a folder
1. Create and open `a.cpp`
1. Paste in the above code
1. Set strict Intellisense - full `c_cpp_properties.json` as follows&hellip;
~~~json
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [],
            "compilerPath": "/usr/bin/clang",
            "cStandard": "c11",
            "cppStandard": "c++17",
            "intelliSenseMode": "clang-x64"
        }
    ],
    "version": 4
}
~~~

Context:
~~~no-highlight
C/C++ for Visual Studio Code v0.20.1

Version: 1.29.0
Commit: 5f24c93878bd4bc645a4a17c620e2487b11005f9
Date: 2018-11-12T07:42:27.562Z
Electron: 2.0.12
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64

Ubuntu 18.04.1 LTS
Linux 4.15.0-39-generic #42-Ubuntu SMP Tue Oct 23 15:48:01 UTC 2018
~~~

Thanks again.

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. It's a bug with our parser (shared with Visual Studio). We'll update this if we get any ETA or fix.

**Comment by tonyelewis** [comment]
Great. Thanks very much for this.

**Comment by sean-mcmanus** [comment]
I think I reported this internally earlier, but I have filed a public issue at https://developercommunity.visualstudio.com/content/problem/456437/c-intellisense-gcc-mode-wont-allow-builtin-expect.html so it can be upvoted.

**Comment by sean-mcmanus** [comment]
The fix for the incorrect IntelliSense error is in the latest Insiders: https://github.com/Microsoft/vscode-cpptools/releases/tag/0.22.2-insiders

**Comment by tonyelewis** [comment]
Great. Thanks very much for all work on this. I'm not on the insiders stream but I look forward to seeing this in the next release.

**Comment by sean-mcmanus** [comment]
The fix is in 0.23.0 now.

**Comment by tonyelewis** [comment]
Yep - this all looks good to me. Thanks very much for all your work on this.
